libgalois/include/katana/PropertyManager.h

```diff
@@ -0,0 +1,64 @@
+#pragma once
+
+#include <cstdint>
+#include <string>
+
+#include "katana/Manager.h"
+#include "katana/MemoryManager.h"
+
+/// Managers tend to the memory consumption for specific, large resources, e.g.,
+/// properties or views.  They interact with the central MemoryManager (MM singleton)
+/// to coordinate memory use.
+
+class PropertyManager : public Manager {
+public:
+  /// Returns the coarse category of memory use
+  ///   e.g., property for the property manager
+  virtual const std::string& MemoryCategory() { return "property"; }
+
+  /// Free standby memory, attempting to free \p goal bytes.
+  /// Returns the number of bytes freed, which can only be less than goal if the
+  /// manager's standby total is less than goal.
+  virtual count_t FreeStandbyMemory(count_t goal);
+
+  // void Insert(
+  //     const katana::Uri& cache_key, const std::shared_ptr<arrow::table>& props);
+
+  void AddProperty(const katana::Uri& property_path) {
+    auto property = cache_->Get(property_path);
+    if (property.has_value()) {
+      auto bytes = katana::ApproxTableMemUse(property.value());
+      MM().StandbyToActive(this, bytes);
+      property_cache_->Evict(property_path);
+    }
+  }
```

Comment:
Shouldn't this return a bool to let users know whether the property was loaded from the cache or not?

IIUC if AddProperty() doesn't succeed, then users load the property explicitly which calls PropertyLoadedCallback(). Is that so? Or is the property cache expected to be shared with the loader?

Reply:
You are right, but I'm not sure if it should return a bool or the `shared_ptr `to the property table.

It turns out that giving all DistPropertyGraphs a storage location took so long I never hooked up the new property cache manager.  But I'd like to do that on the next PR.